patroni/api.py

```diff
@@ -750,6 +753,27 @@ def shutdown_request(self, request):
             _, request = request  # SSLSocket
         return super(RestApiServer, self).shutdown_request(request)
 
+    def get_certificate_serial_number(self):
+        from cryptography import x509, hazmat
+        try:
+            with open(self.__ssl_options['certfile'], "rb") as f:
+                crt = x509.load_pem_x509_certificate(f.read(), hazmat.backends.default_backend())
+                return crt.serial_number
+        except EnvironmentError:
+            logger.exception("Failed to get serial number from certificate %s", self.__ssl_options['certfile'])
```

Comment:
I don't like pulling unnecessary dependencies, they are heavy and also create stress for packagers.

Below is an alternative that relies only on the `ssl` module. 
```suggestion
        if self.__ssl_options.get('certfile'):
            import ssl
            try:
                crt = ssl._ssl._test_decode_cert(self.__ssl_options['certfile'])
                return crt.get('serialNumber')
            except ssl.SSLError as e:
                logger.error('Failed to get serial number from certificate %s: %r', self.__ssl_options['certfile'], e)
```

Yes, I know, `ssl._ssl._test_decode_cert()` is undocumented, but it works on all versions of python.